.pipelines/e2e-step-template.yaml

```diff
@@ -0,0 +1,27 @@
+parameters:
+  job: ''
+
+steps:
+  - script: |
+      mkdir -p '$(GOBIN)'
+      mkdir -p '$(GOPATH)/pkg'
+      mkdir -p '$(modulePath)'
+      shopt -s extglob
+      mv !(gopath) '$(modulePath)'
+      echo '##vso[task.prependpath]$(GOBIN)'
+      echo '##vso[task.prependpath]$(GOROOT)/bin'
+    displayName: 'Set up the Go workspace'
+  - script: make build-binary
+    displayName: Compile
+    workingDirectory: $(modulePath)
+  - script: |
+      export CLIENT_ID=$(SERVICE_PRINCIPAL_CLIENT_ID_E2E_KUBERNETES)
```

Comment:
AZURE_CLIENT_ID?

Reply:
Same here: this is defined in by our e2e tests https://github.com/Azure/aks-engine/blob/master/test/e2e/azure/cli.go#L35 and is the same as circle uses